client/src/unifyfs.c

```diff
@@ -1126,6 +1136,30 @@ int unifyfs_gfid_read_reqs(read_req_t* in_reqs, int in_count)
             }
             delegator_signal();
         }
+
+        /* if this was mread, track the progress */
+        if (count > 1 && !rpc_done) {
+            tmp_rc = unifyfs_mread_rpc_status_check(&mread_ctx);
```

Comment:
Can there be a race condition in here where the ```done``` flag might get set (because the server responds with the data) before we get the return code from the mread rpc and set ```rpc_done```?  If so, we could add a check for rpc_done after this loop and check the status until we get the answer back.

Reply:
@adammoody I see your point. In fact, I did put the rpc checking routine after this loop in my first try. But, the problem was that there were times that the loop wouldn't finish if the server somehow failed to send all data as expected. The reason that I check rpc reply inside this loop is to release the client from checking the data that has failed anyway.

Reply:
I will probably register an issue about remaining issues on client/server coordination for handling read.